### PR TITLE
Fix pointer check

### PR DESCRIPTION
**Pull request by pan-**
Use NULL instead of false in GattClient example.

### CODE REVIEW

**Comment by pan-** [comment]
@hugueskamba Please review.

File: BLE_GattClient/source/main.cpp [hunk 1]
```diff
@@ -463,7 +463,7 @@ class GattClientProcess : private mbed::NonCopyable<GattClientProcess>,
         DiscoveredCharacteristicNode* new_node =
             new(std::nothrow) DiscoveredCharacteristicNode(*characteristic);
 
-        if (new_node == false) {
+        if (new_node == NULL) {
```
Comment (hugueskamba) [inline BLE_GattClient/source/main.cpp]: Why not `nullptr`?

Comment (pan-) [reply BLE_GattClient/source/main.cpp]: good question, my mind keep switching between C++03 and user revision 😅 . Let me change that.

Comment (pan-) [reply BLE_GattClient/source/main.cpp]: I know why I use NULL now. The rest of the file uses NULL so I wanted to keep consistency within the file.

**Pull request merged**